lib/tasks/reporting.rake

```diff
@@ -13,16 +13,50 @@ namespace :reporting do
     Reports::PublishedAttachmentsReport.new.report
   end
 
-  desc "Prints a list of content IDs that documents whose live edition contains a given regular expression"
+  desc "Prints a list of content IDs for documents whose govspeak content contains a given regular expression"
   task :matching_docs, [:regex] => :environment do |_, args|
     regex = Regexp.new(/#{args[:regex]}/)
 
     Document.where.not(live_edition_id: nil).find_in_batches(batch_size: 1000) do |batch|
-      batch.each do |document|
-        next unless document.editions.published.any?
+      batch.each do |object|
+        next unless object.editions.published.any?
 
-        puts document.content_id if regex.match?(document.editions.published.last.body)
+        document_includes_regex(regex, object.content_id, object.class.name, object.editions.published.last.body)
+      end
+    end
+
+    HtmlAttachment.find_in_batches(batch_size: 1000) do |batch|
+      batch.each do |object|
+        document_includes_regex(regex, object.content_id, object.class.name, object.govspeak_content.body)
+      end
+    end
+
+    Person.find_in_batches(batch_size: 1000) do |batch|
+      batch.each do |object|
+        document_includes_regex(regex, object.content_id, object.class.name, object.biography)
+      end
+    end
+
+    PolicyGroup.find_in_batches(batch_size: 1000) do |batch|
+      batch.each do |object|
+        document_includes_regex(regex, object.content_id, object.class.name, object.description)
+      end
+    end
+
+    WorldLocationNews.find_in_batches(batch_size: 1000) do |batch|
+      batch.each do |object|
+        document_includes_regex(regex, object.content_id, object.class.name, object.mission_statement)
+      end
+    end
+
+    WorldwideOffice.find_in_batches(batch_size: 1000) do |batch|
```

Comment:
I'm not sure if it's as easy for Documents and Editions, but could these be a db query instead?
```ruby
WorldwideOffice.where("access_and_opening_times REGEXP ?", '#{regex}').find_each { |d| puts d.content_id }
```

Would that be significantly faster, or not really?

Reply:
I didn't see any change in speed (there's so few records anyway) for Person, PolicyGroup, WorldLocationNews and WorldwideOffice.  It may speed up for Document and HtmlAttachment, but the query would likely become complex as I'd need to include some joins.